src/components/panel/panel.spec.js

```diff
@@ -105,6 +108,62 @@ describe('$mdPanel', function() {
     expect(PANEL_EL).not.toExist();
   });
 
+  it('should call onDomAdded if provided when adding the panel to the DOM', function() {
+    var onDomAdded = jasmine.createSpy();
+    var animationHookConfig = angular.extend({'onDomAdded': onDomAdded}, DEFAULT_CONFIG);
+
+    panelRef = $mdPanel.create(animationHookConfig);
+
+    panelRef.attachOnly().then(function() {
+      expect(onDomAdded).toHaveBeenCalled();
+      expect(PANEL_EL).toExist();
+    });
+  });
+
+  it('should call onOpenComplete if provided after adding the panel to the DOM and animating', function() {
+    var onOpenComplete = jasmine.createSpy();
+    var animationHookConfig = angular.extend({'onOpenComplete': onOpenComplete}, DEFAULT_CONFIG);
+
+    panelRef = $mdPanel.create(animationHookConfig);
+
+    openPanel();
+
+    expect(onOpenComplete).toHaveBeenCalled();
+    expect(PANEL_EL).toExist();
+  });
+
+  it('should call onRemoving if provided when removing the panel from the DOM', function() {
+    var onRemoving = jasmine.createSpy();
+    var onDomRemoved = jasmine.createSpy();
+    var animationHookConfig = angular.extend({'onRemoving': onRemoving, 'onDomRemoved': onDomRemoved}, DEFAULT_CONFIG);
+
+    panelRef = $mdPanel.create(animationHookConfig);
+
+    openPanel();
+
+    closePanel().then(function() {
+      expect(onDomRemoved).not.toHaveBeenCalled();
+      expect(onRemoving).toHaveBeenCalled();
+      expect(PANEL_EL).toExist();
+    });
+
+  });
+
+  it('should call onDomRemoved if provided when removing the panel from the DOM', function() {
```

Comment:
You also needs tests to make sure you callbacks are called when clicking outside the panel to close or when hitting escape to close. We want to make sure the user is able to call functions in those cases, for instance, to redirect focus somewhere on the page. (You can copy the setup of the `clickOutsideToClose` and `escapeToClose` tests.)